### PR TITLE
net: separate acceptor thread from the other net threads

### PR DESCRIPTION
**Pull request by shaitan**

### CODE REVIEW

File: library/pool.c [hunk 1]
```diff
@@ -1181,26 +1217,16 @@ int dnet_io_init(struct dnet_node *n, struct dnet_config *cfg)
...
+		if (err) {
+			goto err_out_protocol_io_stop;
 		}
+	}
 
-		fcntl(nio->epoll_fd, F_SETFD, FD_CLOEXEC);
-		fcntl(nio->epoll_fd, F_SETFL, O_NONBLOCK);
-
-		err = pthread_create(&nio->tid, NULL, dnet_io_process_network, nio);
+	for (i=0; i<n->io->net_thread_num; ++i) {
```
Comment (sabramkin) [inline library/pool.c]:
```suggestion
	for (i = 0; i < n->io->net_thread_num; ++i) {
```

File: library/pool.c [hunk 2]
```diff
@@ -1210,10 +1236,14 @@ int dnet_io_init(struct dnet_node *n, struct dnet_config *cfg)
...
-		pthread_join(n->io->net[i].tid, NULL);
-		close(n->io->net[i].epoll_fd);
+		dnet_net_io_cleanup(&n->io->net[i]);
 	}
 
+	if (n->flags & DNET_CFG_JOIN_NETWORK) {
+		dnet_net_io_cleanup(&n->io->acceptor);
+	}
+err_out_protocol_io_stop:
+	n2_native_protocol_io_stop(n);
```
Comment (sabramkin) [inline library/pool.c]: Thanks.

File: library/pool.c (hunk 1 above)
Comment (sabramkin) [reply library/pool.c]: upd. I see you just moved the code. Ok.

File: library/pool.c [hunk 3]
```diff
@@ -1242,8 +1272,11 @@ void dnet_io_stop(struct dnet_node *n) {
 	dnet_set_need_exit(n);
 
 	for (i = 0; i < io->net_thread_num; ++i) {
-		pthread_join(io->net[i].tid, NULL);
-		close(io->net[i].epoll_fd);
+		dnet_net_io_cleanup(&io->net[i]);
+	}
+
+	if (n->flags & DNET_CFG_JOIN_NETWORK) {
```
Comment (sabramkin) [inline library/pool.c]: Didn't find occurrence of this flag in "-" diff. Seems as something new. Is this flag unset only for client nodes?

Comment (shaitan) [reply library/pool.c]: Yes, this flag is unset for all client nodes.

**Pull request merged**